cypress/integration/7-crypto/crypto.spec.ts

```diff
@@ -14,73 +14,147 @@ See the License for the specific language governing permissions and
+            .should("have.descendants", ".mx_EventTile_receiptSent");
+
+        // Bob sends a response
+        cy.get<Room>("@bobsRoom").then((room) => {
+            bob.sendTextMessage(room.roomId, "Hoo!");
+        });
+        cy.contains(".mx_EventTile_body", "Hoo!")
+            .closest(".mx_EventTile")
+            .should("not.have.descendants", ".mx_EventTile_e2eIcon_warning");
+    };
+
+    const bobJoin = () => {
+        cy.botJoinRoomByName(bob, "Alice").as("bobsRoom");
+        cy.contains(".mx_TextualEvent", "Bob joined the room").should("exist");
+    };
+
+    const bobDoVerification = (request: VerificationRequest): Chainable<EmojiMapping[]> => {
+        return cy.wrap(new Promise<EmojiMapping[]>((resolve) => {
+            const onShowSas = (event: ISasEvent) => {
+                resolve(event.sas.emoji);
+                verifier.off("show_sas", onShowSas);
+                event.confirm();
+                verifier.done();
+            };
+
+            const verifier = request.beginKeyVerification("m.sas.v1");
+            verifier.on("show_sas", onShowSas);
+            verifier.verify();
+        }));
+    };
+
+    const verify = () => {
+        const bobsVerificationRequestPromise = waitForVerificationRequest(bob);
+
+        openRoomInfo().within(() => {
+            cy.get(".mx_RoomSummaryCard_icon_people").click();
+            cy.contains(".mx_EntityTile_name", "Bob").click();
+            cy.contains(".mx_UserInfo_verifyButton", "Verify").click();
+            cy.contains(".mx_AccessibleButton", "Start Verification").click();
+            cy.wrap(bobsVerificationRequestPromise).then((verificationRequest: VerificationRequest) => {
+                verificationRequest.accept();
+                return verificationRequest;
+            }).as("bobsVerificationRequest");
+            cy.contains(".mx_AccessibleButton", "Verify by emoji").click();
+            cy.get<VerificationRequest>("@bobsVerificationRequest").then((request: VerificationRequest) => {
+                return bobDoVerification(request).then((emojis: EmojiMapping[]) => {
+                    emojis.forEach((emoji: EmojiMapping) => {
+                        cy.contains(".mx_VerificationShowSas_emojiSas_emoji", emoji[0]).should("exist");
+                        cy.contains(".mx_VerificationShowSas_emojiSas_label", emoji[1], { matchCase: false })
+                            .should("exist");
+                    });
+                });
+            });
+            cy.contains(".mx_AccessibleButton", "They match").click();
+            cy.contains("You've successfully verified Bob!").should("exist");
+            cy.contains(".mx_AccessibleButton", "Got it").click();
+        });
+    };
+
     beforeEach(() => {
-        cy.startSynapse("default").as('synapse').then(
-            synapse => cy.initTestUser(synapse, "Alice"),
-        );
+        cy.startSynapse("default").then(data => {
+            synapse = data;
```

Comment:
The Cypress docs are pretty insistent that this is an anti-pattern, and that the correct way of doing this is to use aliases. I know this is the approach we've been using in a lot of these tests, but are we sure it's the right way of handling this?

Reply:
I think you are referring to this: https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Aliases I will try to implement it the way the cypress docs suggest it.